cli/src/config.rs

```diff
@@ -196,12 +196,7 @@ pub fn lockfiles(
             let project_lockfiles = project.map(|project| project.lockfiles());
 
             // Fallback to walking the directory.
-            let lockfiles = project_lockfiles.unwrap_or_else(|| {
-                phylum_lockfile::find_lockfiles()
-                    .drain(..)
-                    .map(|(path, format)| LockfileConfig::new(path, format.to_string()))
-                    .collect()
-            });
+            let lockfiles = project_lockfiles.unwrap_or_else(|| find_lockfiles("."));
```

Comment:
We use "." as the search root here, though technically `project.root` would be correct. However the project's lockfiles are initialized to an empty array so they're always provided.

Because of this there's no way to differentiate between an empty lockfiles array and a missing one, so we need to treat both the same and not do recursive lockfile search.

Reply:
> Because of this there's no way to differentiate between an empty lockfiles array and a missing one, so we need to treat both the same and not do recursive lockfile search.

If I'm following correctly, this is when a `.phylum_project` file has been located, but it does not contain any lockfiles. Why is it better to error than to do a recursive search in that case? Even if they explicitly created the `.phylum_project` file without any lockfiles, it still seems like a recursive search is reasonable.

Reply:
> If I'm following correctly, this is when a .phylum_project file has been located, but it does not contain any lockfiles. Why is it better to error than to do a recursive search in that case? Even if they explicitly created the .phylum_project file without any lockfiles, it still seems like a recursive search is reasonable.

The lockfiles field is `#[serde(default)] Vec<String>`. So there is no "located, but does not contain any lockfiles".

This closure is only called when there is no project file.

Reply:
> That's exactly my issue with it. A project file with an empty list (or no list) causes a guaranteed error because we don't do the recursive search.

Correct.

> We should probably also do a recursive search if the list is empty. That way we stay consistent and only error if the recursive search doesn't find any lockfiles.

I disagree. That would stop people from blacklisting all lockfiles in their repositories. Having a project file without lockfile right now is not supported, if we want to change that it should be done in a different PR. The expectation right now is that every project file has a list of lockfiles, if that list is empty, it means there are no lockfiles.

Reply:
> Having a project file without lockfile right now is not supported, if we want to change that it should be done in a different PR.

This is a good point. This PR is probably not the right place to argue about what a user intends when they do some unsupported thing.